### ISSUE TITLE
e4b tools error compiling project with config_setting that controls a bind rule to an alias rule

### ISSUE THREAD

**Issue by psaville-google** [bazelbuild/e4b#8]
I pulled a local bazel repo clone and built bazel from HEAD yesterday in order to test if https://github.com/bazelbuild/e4b/issues/6 fixed my issue with e4b builds.

Sadly, it still doesn't work - see Issue 6 for WORKSPACE and BUILD files for issue reproduction.

```
*** Running [/usr/local/bin/bazel, --watchfs, build, --package_path, %workspace%:/Users/psaville/eclipse/java-neon/Eclipse.app/Contents/Eclipse/../../../../../.p2/pool/plugins/com.google.devtools.bazel.e4b_0.0.1.201604271316/resources, --experimental_interleave_loading_and_analysis, --strategy=Javac=worker, --noexperimental_check_output_files, --aspects=tools/must/be/unique/e4b_aspect.bzl%e4b_aspect, //java/..., //javatests/..., //example/java/..., //scripts/testing/tee:setup_tee, //production/kubernetes/..., //example/java/..., //scs/rapidsync:all] from /Users/psaville/git-gerrit/skybox3 ***
WARNING: Option 'experimental_interleave_loading_and_analysis' is deprecated.
____Loading...
ERROR: /private/var/tmp/_bazel_psaville/33cf99881007ceae82889b09c83f121b/external/maven_prodx_libprodx/jar/BUILD:2:1: in //tools/must/be/unique:e4b_aspect.bzl%e4b_aspect aspect on java_import rule @maven_prodx_libprodx//jar:jar: 
Traceback (most recent call last):
    File "/private/var/tmp/_bazel_psaville/33cf99881007ceae82889b09c83f121b/external/maven_prodx_libprodx/jar/BUILD", line 2
        //tools/must/be/unique:e4b_aspect.bzl%e4b_aspect(...)
    File "/Users/psaville/.p2/pool/plugins/com.google.devtools.bazel.e4b_0.0.1.201604271316/resources/tools/must/be/unique/e4b_aspect.bzl", line 101, in _aspect_impl
        (java_rule_ide_info, java_ide_resol...) = java_rule_ide_info(target, ctx)
Variable 'java_rule_ide_info' is referenced before assignment. The variable is defined in the global scope..
ERROR: /private/var/tmp/_bazel_psaville/33cf99881007ceae82889b09c83f121b/external/maven_prodx_flags/jar/BUILD:2:1: in //tools/must/be/unique:e4b_aspect.bzl%e4b_aspect aspect on java_import rule @maven_prodx_flags//jar:jar: 
Traceback (most recent call last):
    File "/private/var/tmp/_bazel_psaville/33cf99881007ceae82889b09c83f121b/external/maven_prodx_flags/jar/BUILD", line 2
        //tools/must/be/unique:e4b_aspect.bzl%e4b_aspect(...)
    File "/Users/psaville/.p2/pool/plugins/com.google.devtools.bazel.e4b_0.0.1.201604271316/resources/tools/must/be/unique/e4b_aspect.bzl", line 101, in _aspect_impl
        (java_rule_ide_info, java_ide_resol...) = java_rule_ide_info(target, ctx)
Variable 'java_rule_ide_info' is referenced before assignment. The variable is defined in the global scope..
ERROR: Analysis of aspect '/tools/must/be/unique/e4b_aspect.bzl%e4b_aspect of //example/java:ExampleServer' failed; build aborted.
```

**Comment by damienmg** [comment]
Ok I think the issue is because java_rule_ide_info is the name of a function and the name of a local variable, we should just rename that local variable in the skylark aspect. 

https://github.com/bazelbuild/e4b/blob/master/com.google.devtools.bazel.e4b/resources/tools/must/be/unique/e4b_aspect.bzl#L101 for reference

**Comment by psaville-google** [comment]
I renamed the output local variable to from line 101.
FROM: java_rule_ide_info
TO: java_rule_ide__info
UPDATED: line 107

My project now builds... This is pretty great.
Is there any chance you can create a quick patch? I'm too lazy to follow the CLA process, but if there is an internal shortcut, let me know. Should be easy enough to locate my @google.com email.

Thanks.

**Comment by damienmg** [comment]
No shortcut :( but you can set your git author to your google.com email and
the cla should be ok.

**Comment by psaville-google** [comment]
I've created pull request #9.
https://github.com/bazelbuild/e4b/pull/9

**Comment by damienmg** [comment]
I pushed the change to gh-pages and released this version as 0.0.1, Thanks for your help!

**Comment by psaville-google** [comment]
No problem.
